### PR TITLE
performance improvement for split_libraries_fastq.py

### PR DESCRIPTION
**Pull request by gregcaporaso**

### CODE REVIEW

**Comment by qiime-r2d2** [comment]
Build results will soon be (or already are) available at: http://ci.qiime.org/job/qiime-github-pr/1538/

File: qiime/split_libraries_fastq.py [hunk 1]
```diff
@@ -55,24 +55,45 @@ def bad_chars_from_threshold(first_bad_char):
...
+    # we'll shift the index by 1 to the right.
+    idx += 1
+
+    if condition[0]:
+        # If the start of condition is True prepend a 0
+        idx = np.r_[0, idx]
+
+    if condition[-1]:
+        # If the end of condition is True, append the length of the array
+        idx = np.r_[idx, condition.size] # Edit
```
Comment (wasade) [inline qiime/split_libraries_fastq.py]: `# Edit` comment?

File: qiime/split_libraries_fastq.py [hunk 2]
```diff
@@ -55,24 +55,45 @@ def bad_chars_from_threshold(first_bad_char):
...
+    d = np.diff(condition)
+    idx, = d.nonzero()
+
+    # We need to start things after the change in "condition". Therefore,
+    # we'll shift the index by 1 to the right.
+    idx += 1
+
+    if condition[0]:
+        # If the start of condition is True prepend a 0
+        idx = np.r_[0, idx]
```
Comment (wasade) [inline qiime/split_libraries_fastq.py]: did not know about `r_` or `c_`, that's pretty cool

**Comment by wasade** [comment]
:+1:, just a small comment

**Comment by qiime-r2d2** [comment]
Build results will soon be (or already are) available at: http://ci.qiime.org/job/qiime-github-pr/1542/

**Pull request merged**